Tests/test_file_gif.py

```diff
@@ -495,6 +495,42 @@ def test_identical_frames(self):
         # Assert that the new duration is the total of the identical frames
         self.assertEqual(reread.info["duration"], 4500)
 
+    def test_totally_identical_frames(self):
+        duration_list = [1000, 1500, 2000, 4000]
+
+        out = self.tempfile("temp.gif")
+
+        image_path = "Tests/images/bc7-argb-8bpp_MipMaps-1.png"
+        im_list = [
+            Image.open(image_path),
+            Image.open(image_path),
+            Image.open(image_path),
+            Image.open(image_path),
+        ]
+        mask = Image.new("RGBA", im_list[0].size, (255, 255, 255, 0))
+
+        frames = []
+        for image in im_list:
+            frames.append(Image.alpha_composite(mask, image))
+
+        # duration as list
+        frames[0].save(
+            out,
+            save_all=True,
+            append_images=frames[1:],
+            optimize=False,
+            duration=duration_list,
+            loop=0,
+            transparency=0,
+        )
+        reread = Image.open(out)
+
+        # Assert that the first three frames were combined
```

Comment:
'first three frames' - is this correct? Are there not four frames that are combined into one?

Reply:
fixed

Reply:
Thanks